src/main/java/com/bloxbean/cardano/client/crypto/bip32/HdKeyGenerator.java

```diff
@@ -206,6 +212,58 @@ public HdKeyPair getChildKeyPair(HdKeyPair parent, long child, boolean isHardene
         return key;
     }
 
+    /**
+     * Derive the public child key from HD parent public key
+     *
+     * @param parent     the parent key
+     * @param child      the child index
+     * @return
+     */
+    public HdPublicKey getChildPublicKey(HdPublicKey parent, long child) {
```

Comment:
Can we use "int" type instead of "long" for child parameter ?
As per the spec, for soft index =  i < 2^31. For hardened index, i ≥ 2^31

As pub key -> pub key derivation is only applicable for soft index, I think we should be ok with "int" type as the max value of int is 2,147,483,648 (2 ^31)

Reply:
I have switched to int child.